crates/bitwarden-crypto/src/xchacha20.rs

```diff
@@ -20,19 +20,30 @@ use rand::{CryptoRng, RngCore};
 
 use crate::CryptoError;
 
-#[allow(unused)]
 pub(crate) struct XChaCha20Poly1305Ciphertext {
     nonce: GenericArray<u8, <XChaCha20Poly1305 as AeadCore>::NonceSize>,
-    ciphertext: Vec<u8>,
+    encrypted_bytes: Vec<u8>,
 }
 
-#[allow(unused)]
-fn encrypt_xchacha20_poly1305(
+impl XChaCha20Poly1305Ciphertext {
+    pub(crate) fn nonce(&self) -> &[u8; 24] {
+        self.nonce
+            .as_slice()
+            .try_into()
+            .expect("Nonce size is 24 bytes")
+    }
```

Comment:
If we remove the reference from here, we can do into() and get type checked length matching:

```suggestion
    pub(crate) fn nonce(&self) -> [u8; 24] {
        self.nonce.into()
    }
```